dspace-api/src/main/java/org/dspace/identifier/VersionedDOIIdentifierProvider.java

```diff
@@ -0,0 +1,350 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.identifier;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.dspace.identifier.doi.DOIConnector;
+import org.dspace.identifier.doi.DOIIdentifierException;
+import org.dspace.services.ConfigurationService;
+import org.dspace.versioning.Version;
+import org.dspace.versioning.VersionHistory;
+import org.dspace.versioning.service.VersionHistoryService;
+import org.dspace.versioning.service.VersioningService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Required;
+
+/**
+ *
+ * @author Marsa Haoua
+ * @author Pascal-Nicolas Becker (dspace at pascal dash becker dot de)
+ */
+public class VersionedDOIIdentifierProvider extends DOIIdentifierProvider
+{
+ /** log4j category */
+    private static Logger log = Logger.getLogger(VersionedDOIIdentifierProvider.class);
+    
+    private DOIConnector connector;
+
+    static final char DOT = '.';
+    private static final String pattern = "\\d+\\" + String.valueOf(DOT) +"\\d+";
```

Comment:
Make this variable protected, so that if custom code is needed the class inheriting from this one can make use it.